packages/server/graphql/mutations/changeTaskTeam.ts

```diff
@@ -55,6 +56,7 @@ export default {
     }
 
     // RESOLUTION
+    // filter mentions of old team members from task content
```

Comment:
it's possible that someone calls this mutation outside of our UI.
If that happens, we need this mutation to guarantee that an integration exists on the new `teamId`

I propose the following:
- remove the OAuth2 flow from the client, if they click cancel, we abort, just like you have written. if they click OK, then we call this mutation
- this mutation looks up the AtlassianAuth (or GitHubAuth) for the viewerId/teamId combination
- if it exists, the teamId param is valid
- if it doesn't exist, then it looks up the Auth for the viewerId/oldTeamId
- if that doesn't exist, return error: No valid integration auth
- If that does exist, then copy that row & insert it with the updated teamId

eliminating the extra oauth2 flow means the process goes faster & more importantly we can verify that the new teamId supports that task integration

Reply:
I got a bit lost in edge cases. When the viewer has no integration in the target team, there are these options:
1) if user has integration in source team, then we will use that, but still ask
2) if accessUser is someone else and they have integration for target team, then we will use that without asking
3) else we need to ask the user to integrate with complete oauth flow

For now I ignore 2) and 3) and just assume it's 1) as that's the most common case in the UI.
Better would be of course to detect 2) and 3) as well and handle it appropriately in the UI, but we do not get all information necessary at the moment via GraphQL and I also don't want to repeat the same logic if there is a nicer solution.

Let's just roll with 1) and fix 2) and 3) another day, user value is probably also minimal. Use case 2) I can imagine is a bit annoying, i.e. team lead has 2 parabol teams and adds integration. They're also the only ones with access to Jira. Every time some other team member would move a task, they would get shown the dialog. It would still move the task, but it might get annoying. But if that hypothetical user exists, they can open a ticket.

Reply:
fantastic work exploring all the edge cases.
i think your assumption are reasonable & i feel very comfortable with the solution you've built!